src/CompilerOptions.ts

```diff
@@ -1,12 +1,28 @@
 import * as ts from "typescript";
 
-export interface CompilerOptions extends ts.CompilerOptions {
+type KnownKeys<T> = {
+    [K in keyof T]: string extends K ? never : number extends K ? never : K
+} extends { [_ in keyof T]: infer U }
+    ? U
+    : never;
+
+type OmitIndexSignature<T extends Record<any, any>> = Pick<T, KnownKeys<T>>;
+
+export interface TransformerConfig {
+    transform: string;
+    when?: keyof ts.CustomTransformers;
+    [option: string]: any;
+}
+
+export type CompilerOptions = OmitIndexSignature<ts.CompilerOptions> & {
     noHeader?: boolean;
     luaTarget?: LuaTarget;
     luaLibImport?: LuaLibImportKind;
     noHoisting?: boolean;
     sourceMapTraceback?: boolean;
-}
+    tsTransformers?: TransformerConfig[];
+    [option: string]: ts.CompilerOptions[string] | TransformerConfig[];
```

Comment:
Not sure I like this index type being here and the complexity to the type it adds, is this really required?

Reply:
Index signature in `ts.CompilerOptions` disallows adding new options with types that aren't defined in it. As a workaround `OmitIndexSignature` removes it from `ts.CompilerOptions` and later we re-add it with our added type.
Re-adding index signature is actually not strictly required there (because `ts.CompilerOptions` still would be a subtype), but it makes sense considering that `tstl.CompilerOptions` is supposed to be just `ts.CompilerOptions` with custom options.